docs/developers/nft-tokens.md

```diff
@@ -0,0 +1,400 @@
+          "@416c696365546f6b656e73" + # AliceTokens
+          "@414c43" +                 # ALC
+}
+```
+
+**Step 2: Fetch the token identifier**
+
+For doing this, one has to check the previously sent transaction and see the Smart Contract Result of it.
+It will look similar to `@ok@414c432d317132773365`. The `414c432d317132773365` represents the token identifier in hexadecimal encoding.
+
+**Step 3: Set roles**
+
+Assign `ESDTRoleNFTCreate` and `ESDTRoleNFTAddQuantity` roles to an address. You can set these roles to your very own address.
+```
+{
+    Sender: <your address>
+    Receiver: erd1qqqqqqqqqqqqqqqpqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqzllls8a5w6u
+    Value: 0
+    GasLimit: 60000000
+    Data: "setSpecialRole" +
+          "@414c432d317132773365" +   # previously fetched token identifier
+          "@" + <address to assign the roles in a hexadecimal encoding> +
+          "@45534454526f6c654e4654437265617465" +  # ESDTRoleNFTCreate
+          "@45534454526f6c654e46544164645175616e74697479" # ESDTRoleNFTAddQuantity
+          ...
+}
+```
+
+**Step 4: Create NFT**
+
+Fill all the attributes as you think.
+
+```
+{
+    Sender: <address with ESDTRoleNFTCreate role>
+    Receiver: <same as sender>
+    Value: 0
+    GasLimit: 60000000
+    Data: "ESDTNFTCreate" +
+          "@414c432d317132773365" +   # previously fetched token identifier
+          "@" + <initial quantity in hexadecimal encoding> +
+          "@" + <NFT name in hexadecimal econding> +
+          "@" + <Royalties in hexadecimal encoding> +
+          "@" + <Hash in hexadecimal encoding> +
+          "@" + <Attributes in hexadecimal encoding> +
+          "@" + <URI in hexadecimal encoding> +
+          "@" + <URI in hexadecimal encoding> +
+}
+```
+
+:::tip
+Note that the nonce is very important when creating an NFT. You must save the nonce after NFT creation because you will need it for further actions.
+For example, if Alice has nonce `5` before sending the nft creation transaction, nonce `6` must be used in further actions.
+Also, the `nonce` can be fetched by viewing all the tokens for the address via API.
+:::
+
+**Step 5: Transfer**
+
+```
+{
+    Sender: <your address>
+    Receiver: <same as sender>
+    Value: 0
+    GasLimit: 500000 + length of Data field
+    Data: "ESDTNFTTransfer" +
+          "@414c432d317132773365" +   # previously fetched token identifier
+          "@" + <the nonce saved above in hexadecimal encoding> + 
+          "@" + <quantity to transfer in hexadecimal encoding> +
+          "@" + <destination address in hexadecimal encoding>
+}
+```
+
+## **Rest API**
+
+There are a number of API endpoints that one can use to interact with ESDT NFT data. These are:
+
+### <span class="badge badge-primary">GET</span> **Get NFT data for an address**
+<!--DOCUSAURUS_CODE_TABS-->
+
+<!--Request-->
+Returns the balance of an address for specific ESDT Tokens.
+
+```
+https://api.elrond.com/address/*bech32Address*/nft/*tokenIdentifier*/nonce/*creation-nonce*
```

Comment:
I would put everything that has \*something\* as \<something\> if it is something that is a variable

Reply:
done